.github/workflows/rust.yml

```diff
@@ -17,4 +17,6 @@ jobs:
     - name: Build
       run: cargo build --verbose
     - name: Run tests
-      run: cargo test --verbose
+      run: cargo test --locked  --no-fail-fast --lib
```

Comment:
Wow I didn't know about these flags, I've searched about what they do (https://doc.rust-lang.org/cargo/commands/cargo-test.html) and it is a great addition!